data-gen/gen-autoconf.py

```diff
@@ -1,35 +1,50 @@
-#!/usr/bin/python
-import sys
-import os
+#!/usr/bin/python3
 import fnmatch
+import os
+
 
 def process(s):
-    print 'Processing %s ...' % s,
-    m4s = dict()
-    pkgs = dict()
+    print('Processing %s ...' % s, end=' ')
+    m4s = {}
+    pkgs = {}
     f = os.popen('lzma -c -d %s.txt.lzma' % s)
-    g = open(s + '.sql', 'w')
+    g = open(s+'.sql', 'w')
     g.write('BEGIN;\n')
-    g.write('CREATE TABLE macros(macro VARCHAR(64) NOT NULL, id_m4 INT NOT NULL);\n')
-    g.write('CREATE TABLE m4s(id_m4 INT PRIMARY KEY NOT NULL, m4 VARCHAR(64) NOT NULL, id_pkg INT NOT NULL);\n')
-    g.write('CREATE TABLE packages(id_pkg INT PRIMARY KEY NOT NULL, package VARCHAR(64) NOT NULL);\n')
+    g.write('CREATE TABLE macros('
+            'macro VARCHAR(64) NOT NULL, '
+            'id_m4 INT NOT NULL);\n')
+    g.write('CREATE TABLE m4s('
+            'id_m4 INT PRIMARY KEY NOT NULL, '
+            'm4 VARCHAR(64) NOT NULL, '
+            'id_pkg INT NOT NULL);\n')
+    g.write('CREATE TABLE packages('
+            'id_pkg INT PRIMARY KEY NOT NULL, '
+            'package VARCHAR(64) NOT NULL);\n')
 
     for line in f:
         line = line.strip().split(' ')
-        if not pkgs.has_key( line[0] ):
-            pkgs[ line[0] ] = len(pkgs)
-            g.write("INSERT INTO packages(id_pkg, package) VALUES(%s, '%s');\n" % (pkgs[ line[0] ], line[0]))
-        if not m4s.has_key( line[0]+line[1] ):
-            m4s[ line[0]+line[1] ] = len(m4s)
-            g.write("INSERT INTO m4s(id_m4, m4, id_pkg) VALUES(%s, '%s', %s);\n" % ( m4s[line[0]+line[1]], line[1], pkgs[ line[0] ]))
-        g.write("INSERT INTO macros(macro, id_m4) VALUES('%s', %s);\n" % (line[2], m4s[line[0]+line[1]]))
+        if line[0] not in pkgs:
+            pkgs[line[0]] = len(pkgs)
+            g.write('INSERT INTO packages(id_pkg, package) '
```

Comment:
I understand that this is strictly speaking equivalent to the py2k version, but this is a known security issue. https://docs.python.org/3/library/sqlite3.html The proper form is
```python
g.write('INSERT INTO packages(id_pkg, package) ' +
             "VALUES(?, '?');\n", pkgs[line[0]], line[0])
```

Reply:
Fair point. Done